src/sagemaker/local/pipeline.py

```diff
@@ -273,8 +273,10 @@ class _TrainingStepExecutor(_StepExecutor):
     """Executor class to execute TrainingStep locally"""
 
     def execute(self):
-        job_name = unique_name_from_base(self.step.name)
         step_arguments = self.pipline_executor.evaluate_step_arguments(self.step)
+        job_name = step_arguments.pop("TrainingJobName", None) or unique_name_from_base(
```

Comment:
1. What is the list of steps that we support with the local executor? No ModelStep, Tuning, or Clarify? See the list here on our docs. https://docs.aws.amazon.com/sagemaker/latest/dg/build-and-manage-access.html under `Apply job prefixes to pipeline instantiations`.

2. I may be missing something obvious, but where is the conditional evaluating based of whether the feature flag is set to true? Looks like this is default popping the name if it exists, if not generating using timestamp. By the way, if I remember correctly with Lighstaber, we ran into issues with timestamps being used in the step names, and we created this helper `unique_name_from_base_uuid4` to use instead. Should we keep that behavior consistent for pipeline created step names?

Reply:
1. Correct. https://docs.aws.amazon.com/sagemaker/latest/dg/pipelines-local-mode.html. ModelStep (with create model args) does not launch jobs, so should not matter.
2. This `def execute(self):` is called during execution time so the job name evaluated by `use_custom_job_prefix` is already in the step Arguments field. This `use_custom_job_prefix` should be evaluated in compile time.
3. The local mode is different. And `job_name = unique_name_from_base(self.step.name)` is existing behavior, if no critical issues reported, we should avoid breaking change.